libvirt/tests/src/virtual_network/qemu/hotplug_nic_with_shutdown_reboot.py

```
@@ -0,0 +1,147 @@
+    4) Ping guest new ip from host
+    5) Pause/resume VM and verify connectivity
+    6) Hotunplug the interface
+    7) Perform shutdown or reboot operation based on test variant
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    vm_name = params.get("main_vm")
+    vm = env.get_vm(vm_name)
+    iface_attrs = eval(params.get("iface_attrs", "{}"))
```

Ruff (0.14.7)

Use of possibly insecure function; consider using `ast.literal_eval`

(S307)